Source reformatting: switch to no line-breaking

zcorpan opened the pull request:
Meta: Preparation for source reformatting

* Add data-noreformat attributes
* Fix double spaces and incorrect wrapping around tags
* Add a Check Formatting workflow
* Update CONTRIBUTING.md

Meta: Source reformatting: switch to no line-breaking

This applies reformahtml 0.1.9. See https://github.com/zcorpan/reformahtml

Fixes #1059.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://whatpr.org/html/11640/acknowledgements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/acknowledgements.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/acknowledgements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/browsers.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/browsers.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/browsers.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/browsing-the-web.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/browsing-the-web.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/browsing-the-web.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/canvas.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/canvas.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/canvas.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/common-dom-interfaces.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/common-dom-interfaces.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/common-dom-interfaces.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/common-microsyntaxes.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/common-microsyntaxes.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/common-microsyntaxes.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/comms.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/comms.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/comms.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/custom-elements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/custom-elements.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/custom-elements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/dnd.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/dnd.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/dnd.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/document-lifecycle.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/document-lifecycle.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/document-lifecycle.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/document-sequences.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/document-sequences.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/document-sequences.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/dom.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/dom.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/dom.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/dynamic-markup-insertion.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/dynamic-markup-insertion.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/dynamic-markup-insertion.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/edits.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/edits.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/edits.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/embedded-content-other.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/embedded-content-other.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/embedded-content-other.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/embedded-content.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/embedded-content.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/embedded-content.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/form-control-infrastructure.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/form-control-infrastructure.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/form-control-infrastructure.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/form-elements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/form-elements.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/form-elements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/forms.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/forms.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/forms.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/grouping-content.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/grouping-content.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/grouping-content.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/iana.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/iana.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/iana.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/iframe-embed-object.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/iframe-embed-object.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/iframe-embed-object.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/image-maps.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/image-maps.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/image-maps.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/imagebitmap-and-animations.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/imagebitmap-and-animations.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/imagebitmap-and-animations.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/images.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/images.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/images.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/index.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/index.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/index.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/indices.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/indices.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/indices.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/infrastructure.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/infrastructure.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/infrastructure.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/input.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/input.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/input.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/interaction.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/interaction.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/interaction.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/interactive-elements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/interactive-elements.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/interactive-elements.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/introduction.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/introduction.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/introduction.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/links.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/links.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/links.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/media.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/media.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/media.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/microdata.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/microdata.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/microdata.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/named-characters.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/named-characters.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/named-characters.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/nav-history-apis.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/nav-history-apis.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/nav-history-apis.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/obsolete.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/obsolete.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/obsolete.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/parsing.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/parsing.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/parsing.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/popover.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/popover.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/popover.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/references.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/references.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/references.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/rendering.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/rendering.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/rendering.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/scripting.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/scripting.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/scripting.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/sections.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/sections.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/sections.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/semantics-other.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/semantics-other.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/semantics-other.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/semantics.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/semantics.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/semantics.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/server-sent-events.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/server-sent-events.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/server-sent-events.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/speculative-loading.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/speculative-loading.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/speculative-loading.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/structured-data.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/structured-data.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/structured-data.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/syntax.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/syntax.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/syntax.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/system-state.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/system-state.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/system-state.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/tables.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/tables.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/tables.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/text-level-semantics.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/text-level-semantics.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/text-level-semantics.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/timers-and-user-prompts.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/timers-and-user-prompts.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/timers-and-user-prompts.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/urls-and-fetching.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/urls-and-fetching.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/urls-and-fetching.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/web-messaging.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/web-messaging.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/web-messaging.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/webappapis.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/webappapis.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/webappapis.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/webstorage.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/webstorage.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/webstorage.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/workers.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/workers.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/workers.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/worklets.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/worklets.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/worklets.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )
<a href="https://whatpr.org/html/11640/xhtml.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">/xhtml.html</a>  ( <a href="https://whatpr.org/html/11640/4ce63af...ef51b44/xhtml.html" title="Last updated on Oct 23, 2025, 7:36 AM UTC (ef51b44)">diff</a> )

domenic commented:
Very cool!

The other major section to not reformat is the acknowledgements section.

Is there any way to do some sort of programmatic check that only whitespace has changed, to rule out any content-deleting bugs in the reformatter? Maybe just the htmldiff tool would work?

jmdyck commented:
In the paragraph that begins "Here is an example of color space conversion", the status quo:
   `... <code data-x=\n   "dom-....`
is converted to:
   `... <code data-x= "dom-...`
but that space after the `=` probably shouldn't be there.

(This is the only case where the status quo has a line-break after an attribute's `=`.)

zcorpan committed:
Meta: Preparation for source reformatting

* Add data-noreformat attributes
* Fix double spaces and incorrect wrapping around tags
* Add a Check Formatting workflow
* Update CONTRIBUTING.md

annevk commented:
As far as I can tell we only have one instance. We could fix it separately or incorporate it in a random PR. I don't think it matters much?

annevk commented:
@zcorpan it seems this branch still has conflicts?

annevk reviewed (approved):
Rubber stamp, assuming this is going to be followed up with a "Meta:" commit to fix blame and hoping that this doesn't introduce new issues.

jmdyck commented:
- A lot of the `<ref>` tags are preceded by 2-10 spaces, where it should presumably be only 1.
- Sometimes a `<ref>` isn't joined to the previous line (e.g. lines 3170, 6087), but probably should be.
- Sometimes a `</div>` tag is joined to the previous line (e.g. line 1289), but should probably stay on its own line.
- The treatment of comments seems inconsistent: sometimes joined up, sometimes not. Is there a rule?

zcorpan commented:
I'm happy with the result now. I've looked through the source and added some more `data-noreformat` attributes. I've also added a workflow step to run the script so future changes use correct wrapping. I've also updated the documentation in CONTRIBUTING.md.

zcorpan commented:
I noticed multiple spaces before comments.

jmdyck commented on CONTRIBUTING.md:
````diff
@@ -53,28 +53,9 @@ Due to the long legacy of the existing text, these guidelines are not always app
-  <p>A <code>base</code> element that is the first <code>base</code> element with an <code
-  data-x="attr-base-href">href</code> content attribute <span>in a document tree</span> has a
-```
+You can run https://github.com/zcorpan/reformahtml locally to remove inter-paragraph line wrapping. This script is also run as a GitHub workflow.
````
```suggestion
You can run https://github.com/zcorpan/reformahtml locally to remove intra-paragraph line wrapping. This script is also run as a GitHub workflow.
```

jmdyck commented:
I'm puzzled about the semantics of `data-noreformat`.

E.g., in `source` at line 2449 (under "terms defined in URL"), there's a `<ul>` marked `data-noreformat`, so I expected that the whole of that element (down to line 2509) would be identical in the reformatted file. But there are 3 spots within the element where a linebreak was eliminated.

(Mind you, I think that's the only example where `data-noreformat` seems to have been disrespected.)

jmdyck commented:
There are some spots where the reformatter has incorrectly dropped a space before a start-tag, e.g.
- line 1324: `sets of<span data-x="plugin">plugins`
- line 6292: `The<dfn id="lazy-load-root-margin">lazy load scroll margin`
- line 8670: `the document is a<span>cookie-averse`

jmdyck commented:
> Yes, comments that are preceded with a newline and followed by a newline are considered structural and left alone (including internal linebreaks). Other comments are reflowed inline.

Thanks.

In the source, there are 4 cases where two comments are immediately adjacent. (Scan for `--><!--`). In each case, neither comment qualifies as structural (because the first isn't followed by a newline and the second isn't preceded by a newline), and so the reformatter reflows them. However, in the 2nd+3rd cases (in "The DataTransferItem Interface"), it would probably be better if they were left as is. So in those two cases, I suggest inserting a newline between the two comments in your 'preparation' commit.

domfarolino commented:
Sorry to jump in late, but I'm still not sure what the value add here is. Is it just lower friction for contributors to not have to worry about line breaks? What is the rule we're trying to enforce after this PR? Is it:
 - you can break wherever you want, and have arbitrarily long paragraphs; or
 - no line breaks within a paragraph are allowed

I think I still prefer the 100 character line break rule, and using a tool like [`specfmt`](https://github.com/domfarolino/specfmt) to keep this specification's source more consistent with other WHATWG specs, whereas it seems this change drives more inconsistency across our specs.

annevk commented:
The rules are not consistent though. Most other WHATWG specifications don't permit breaking on every possible space. Moving towards wrapping in your editor seems acceptable to me and if this works out we can port it to DOM, Fetch, etc.

zcorpan committed:
Meta: Source reformatting: switch to no line-breaking

This applies reformahtml 0.1.9. See https://github.com/zcorpan/reformahtml

Fixes #1059.

domfarolino commented:
Also, is there a way to preserve indentation among other things in GitHub's review panel? Or is the plan to just live with wonky indents like:
<img width="854" height="102" alt="Screenshot 2025-10-14 at 6 09 17 PM" src="https://github.com/user-attachments/assets/b988eeff-9859-4a8b-9f85-464f17c96276" />

I worry about this degrading readability during reviews.

Also, I worry this would make PRs harder to review, since you can't anchor comments in the middle of a visually wrapped line on GitHub. It will be hard to reference specific text in the middle of a paragraph, since your comment will just be anchored at the first visual line of the whole block, which is really weird. _Suggestions_ won't be harder, since you an make modifications in the middle of a block of text (albeit the block will be larger now), but I think the comment experience would be pretty annoying when trying to target very specific text.

jmdyck commented:
> What are some good, recommended editor settings to visually wrap existing long lines to ~100 characters, and preserve the indentation level? Specifically for vim, I can't find a good way to visually wrap long lines tighter than the vim window itself, which leaves the spec pretty unreadable to me. Does anyone know the trick in vim?

```vim
:set wrap linebreak breakindent breakindentopt=shift:1
```
will visually wrap long lines at word-boundaries and indent the wrapped lines at 1 more than the indent of the actual line. But the wrapping-limit is still defined by the width of the window -- I don't think you can visually wrap to a specified width, but I might have missed something.

zcorpan commented:
Pros of no-line-breaks:
* No need for contributors to think about where to put line breaks, or to run a script to normalize line breaks.
* Searching for text is easier, because text is not wrapped across multiple lines. (In VS Code, I need to use `\n?\s+` between words. In some other editors, `\s+` is enough.)
* Diffs in GitHub should more reliably highlight only changed words. When wrapping to 100 cols, sometimes a whole block is marked as modified (without the changed words highlighted).

Cons:
* Some editors might not have satisfactory soft wrapping support.
* GitHub UI's soft wrapping doesn't indent wrapped lines. (This is fixable by GitHub.)
* GitHub code review: it's harder to comment on a specific part of a paragraph. (Maybe this is fixable by GitHub.)

@domfarolino suggested running `specfmt` as part of `html-build`. I think this still introduces friction for contributors. Running `html-build` is often not needed (for small changes I usually don't build locally). `specfmt` also requires changes to be committed before running. The current build system doesn't modify the `source` file. If it's an opt-in to reformat, it seems not so different from the status quo.

zcorpan commented:
This is currently waiting for one of:

- GitHub UI changes (see https://github.com/orgs/community/discussions/192705 )
- Us deciding to go ahead with this change anyway.
